Consolidate and remove unsupported .NET targets

jshergal opened the pull request:
Updated all projects to current SDK style csproj format. Removed outdated/unsupported framework targets.
Removed older .NET Framework targets which are unsupported in MSTest (net20 and net35).

This PR resolves #5

jshergal committed:
Consolidate and remove unsupported .NET targets

Updated all projects to current SDK style csproj format.
Removed outdated/unsupported framework targets.
Removed older .NET Framework targets which are unsupported in MSTest (net20 and net35).

ljnath commented:
We want to support backward compatibility to all available .NET versions, so will skip this PR

jshergal committed:
Added back .NET Framework 2.0 and added unit tests for old frameworks

net20 target supports .NET Framework 2.0 - 3.5.
netstandard20 supports .NET Framework 4.6.1 (with caveats) - 4.8.1
netstandar21 supports .NET Core 1.0 - NET 5
net6.0 supports .NET 6-7
net8.0 supports .NET 8+

jshergal commented:
@ljnath I have updated the PR to include support to all available .NET versions. I added an NUnit unit test project to test the older .NET Frameworks.

Just as a side note, only NUnit has support for testing the older frameworks and even there it is only .NET Framework 3.5. All other testing frameworks require a minimum target of .NET Framework 4.6.1, so you may want to consider doing a "last release" that has support for .NET Framework 2.0 - 4.5, and sunset those targets.

jshergal committed:
Added reference assemblies for .NET Framework 3.5

jshergal committed:
Modified workflow yaml for modern dotnet build

jshergal committed:
Merge branch 'UpdateFrameworks' into WorkflowTest

jshergal committed:
Fixing issue with code coverage? Added manual trigger

jshergal commented:
Hi @ljnath, I believe the build will still fail to run even after these changes. I do have an updated yaml file that you can take a look at. It is in the master branch of my fork. I have run and tested it on my side and while the build will still fail with the current master branch, I could create a separate PR for the build script, or you could just copy it and update it. Once that is done, the build should work for this PR.

jshergal committed:
Updated workflow to use latest release .NET 8 for build

jshergal committed:
Updated testing libraries and fixed YAML file

jshergal committed:
Updated build YAML

jshergal committed:
Updated build YAML to run on push to any branch

jshergal commented:
Hi @ljnath,
Just a quick update here, as I haven't heard anything back from you on any of the comments I left several days ago. Please let me know what you think about this PR and on the idea of you pushing one more release that will support .NET Framework 2.0 and 3.5, before dropping support for those frameworks.

As I pointed out in my other comments, the major Unit Testing libraries (xUnit, NUnit and MSTest) do not support running tests on those frameworks, and so while we could maintain the tag in the code for a build to add the library, it seems like an unnecessary burden to try to continue to support something that only a handful of legacy systems may still need to support.

If we went ahead with a final release that includes support there would still be the code improvement regarding the seeding and such as well as now having moved things internally forward into a more modern tooling approach. With that done, I would be free to go ahead and make the other improvements, etc, which I have in mind.

With that in mind, if I do not hear back, then you are welcome to take this PR or close it, whatever you see fit, but I will go ahead and create a separate NuGet package and take my fork and run with it.

Cheers!

ljnath commented:
Hello @jshergal ,

I apologize for the delayed response. While I understand the limitations of the legacy .NET Framework versions (2.0 & 3.5), I still plan to support them for as long as possible. Given that this is a micro library, I'm open to skipping tests for these frameworks in order to maintain support for these older versions.

I’d love to hear your thoughts on this.

jshergal commented:
Hi @ljnath,
No worries if you want to continue to support .NET Framework 2.0 and 3.5. This PR I have here continues to support those and includes a unit test project that should validate the code at least on .NET Framework 3.5, so feel free to accept it or close it. This PR also includes an updated build YAML that will work with latest GitHub Actions, so if you accept it, it should fix the issue with the builds failing. If you have a question about it, feel free to ask. Hopefully it is a help to you.
